### PR TITLE
[DDW-892] Use fresh locale while rebuilding the application menu

### PR DESCRIPTION
**Pull request by szymonmaslowski**
This PR makes the application top bar menu to update according to currently selected language immediately after the user selects that language.

The issue wast that even though menu was rebuilding on every language change, the locale used for that was outdated. Now every time there is a menu rebuild request current value of locale is accessed.

## Todos

- [ ] Ensure there is no issue on windows or linux

## Screenshots

image
image

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1642081265040400)

---

### Basics

- [x] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [x] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR has default-sized Daedalus window screenshots or animated GIFs of important UI changes:
  - [x] In English
  - [x] In Japanese
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [x] There are no *flow* errors or warnings (`yarn flow:test`)
- [x] There are no *lint* errors or warnings (`yarn lint`)
- [x] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Review by mchappell — APPROVED** [review]
Don't forget the CHANGELOG 👍

**Review by renanvalentin — APPROVED** [review]
Great job ! 🎉

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### Fixes
 
 - Fixed receiver address validation by disallowing rewards addresses ([PR 2781](https://github.com/input-output-hk/daedalus/pull/2781))
+- Fixed immediate language updates of application top menu bar ([PR 2813](https://github.com/input-output-hk/daedalus/pull/2813))
```
Comment (mchappell) [inline CHANGELOG.md:8]: Please put this one at the top of the list for fixes

Comment (szymonmaslowski) [reply CHANGELOG.md:7]: Done :+1:

**Pull request merged**